tests/src/test/scala/org/http4s/ContentCodingSpec.scala

```diff
@@ -16,13 +16,11 @@ class ContentCodingSpec extends Http4sSpec {
   "compare" should {
     "be consistent with coding.compareToIgnoreCase for same quality" in {
       prop { (a: ContentCoding, b: ContentCoding) =>
-        (a.qValue == b.qValue) ==> (a.coding.compareToIgnoreCase(b.coding) must_== a.compare(b))
-      }
-    }
-    "be consistent with qValue.compareTo for same coding" in {
-      prop { (a: ContentCoding, b: ContentCoding) =>
-        (a.coding.toLowerCase == b.coding.toLowerCase) must_== (a.qValue.compareTo(b.qValue) == a
-          .compare(b))
+        ((a.coding, a.qValue), (b.coding, b.qValue)) match {
```

Comment:
Should I add `ContentCoding.unapply`?

Reply:
I think not, because the equality of what gets unapplied is not the same as the equality of the object.  I think it's similar to the argument against the case class.  Where it behaves like a product, I would say yes.  Here, I don't think so.